### PR TITLE
[HEVC][10bit] Support encode 8bit from 10bit for sampe_encode and sample_multitranscode

### PR DESCRIPTION
**Pull request by SofyaBalandina**
Issue: MDP-66051
Test: manually

Usage examples:

sample_encode.exe:
VPP based bit-depth conversion (the conversion will be done due to the different forсс):
h265 -i Cactus_ProRes_1280x720_50f.p010.yuv -o out.hevc -w 1280 -h 720  -lowpower:off -p010 -ec::nv12
Encoder internal bit-depth conversion:
h265 -i Cactus_ProRes_1280x720_50f.p010.yuv -o out.hevc -w 1280 -h 720  -lowpower:on -TargetBitDepthLuma 8 -TargetBitDepthChroma 8 -p010

sample_multi_transcode.exe:
VPP based bit-depth conversion (the conversion will be done due to the different forсс):
-i::h265 off.hevc -f 60 -h 720 -lowpower:off -n 10  -w 1280 -hw_d3d11 -o::h265 out_out_off.h265 -ec::nv12 -dc::p010
Encoder internal bit-depth conversion:
-i::h265 on.hevc -f 60 -w 1280 -h 720 -lowpower:on -n 10 -hw_d3d11 -o::h265 out_out_on.h265 -TargetBitDepthLuma 8 -TargetBitDepthChroma 8

### CODE REVIEW

File: samples/sample_encode/src/pipeline_encode.cpp [hunk 1]
```diff
@@ -545,6 +545,21 @@ mfxStatus CEncodingPipeline::InitMfxEncParams(sInputParams *pInParams)
     m_mfxEncParams.mfx.FrameInfo.PicStruct    = pInParams->nPicStruct;
     m_mfxEncParams.mfx.FrameInfo.Shift        = pInParams->shouldUseShifted10BitEnc;
 
+#if (MFX_VERSION >= 1027)
+    // set BitDepthLuma/BitDepthChroma in case of VME
+    if (!pInParams->enableQSVFF)
```
Comment (dmitryermilov) [inline samples/sample_encode/src/pipeline_encode.cpp:550]: why do we need this?

Comment (SofyaBalandina) [reply samples/sample_encode/src/pipeline_encode.cpp:550]: enableQSVFF is true, if lowpower is on and false, if lowpower is off
When lowpower is off, сonverting is doing with vpp and we should force enc parameters with our request
BitDepthLuma/BitDepthChroma for encoder will be set according to enc forcc
Enc forcc can be specify by user or by app with some default value
So, if the user does not specify enc and input forcc, but specifies TargetBitDepthLuma/TargetBitDepthChroma parameters and we not force setting BitDepthLuma/BitDepthChroma, the encoding can pass, but the result will not be what was expected (BitDepthLuma/BitDepthChroma will be correspond to default value of forcc)

Comment (dmitryermilov) [reply samples/sample_encode/src/pipeline_encode.cpp:550]: Let's talk verbally tomorrow.

**Comment by dmitryermilov** [comment]
@SofyaBalandina , LGTM. But can you please provide two examples of cmd lines **into commit message**:
1) With VPP based bit-depth conversion 
2) With encoder internal bit-depth conversion.  

?

**Comment by dmitryermilov** [comment]
> @SofyaBalandina , LGTM. But can you please provide two examples of cmd lines **into commit message**:
> 
> 1. With VPP based bit-depth conversion
> 2. With encoder internal bit-depth conversion.
>    ?
> 
> Can you please also update https://github.com/Intel-Media-SDK/MediaSDK/blob/master/doc/samples/readme-encode_linux.md accordingly ?

@SofyaBalandina , I see you added into PR description but it's still missing into commit message... So looking at git history a reading won't see these examples.

**Comment by SofyaBalandina** [comment]
> > @SofyaBalandina , LGTM. But can you please provide two examples of cmd lines **into commit message**:
> > 
> > 1. With VPP based bit-depth conversion
> > 2. With encoder internal bit-depth conversion.
> >    ?
> > 
> > Can you please also update https://github.com/Intel-Media-SDK/MediaSDK/blob/master/doc/samples/readme-encode_linux.md accordingly ?
> 
> @SofyaBalandina , I see you added into PR description but it's still missing into commit message... So looking at git history a reading won't see these examples.

sorry, I believed description got into commit message in merge, fixed it (add to commit message to last commit)

**Review by dmitryermilov — APPROVED** [review]

**Pull request merged**